module/VuFindSearch/src/VuFindSearch/Backend/Solr/Backend.php

```diff
@@ -428,12 +432,11 @@ public function workExpressions($id, $workKeys, ParamBag $defaultParams = null)
         $params = $defaultParams ? clone $defaultParams
             : new \VuFindSearch\ParamBag();
         $this->injectResponseWriter($params);
-        $query = [];
+        $terms = [];
         foreach ($workKeys as $key) {
-            $key = addcslashes($key, '+-&|!(){}[]^"~*?:\\/');
-            $query[] = "work_keys_str_mv:(\"$key\")";
+            $terms[] = addcslashes($key, ',+-&|!(){}[]^"~*?:\\/');
         }
-        $params->set('q', implode(' OR ', $query));
+        $params->set('q', '{!terms f=work_keys_str_mv}' . implode(',', $terms));
```

Comment:
Can we guarantee that no term will ever contain a comma?

Reply:
No. I thought escaping the delimiter above would be enough, but now that I checked it, that doesn't work. I'll think about an alternative.

Reply:
I changed the delimiter to chr(1F) and removed the useless escaping (verified manually with Solr that all the characters can be in a term as is).